src/commands/utility/my-points.js

```diff
@@ -0,0 +1,124 @@
+const { SlashCommandBuilder } = require('discord.js');
+const { translateLanguage } = require('../../languages/index');
+const { VOTE_POINTS } = require('../../config');
+
+const tagIds = VOTE_POINTS.TAG_IDS;
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('my-points')
+    .setDescription(translateLanguage('myPoints.description'))
+    .addIntegerOption((option) =>
+      option
+        .setName('year')
+        .setDescription(translateLanguage('myPoints.yearOption'))
+        .setRequired(false)
+    )
+    .addIntegerOption((option) =>
+      option
+        .setName('month')
+        .setDescription(translateLanguage('myPoints.monthOption'))
+        .setRequired(false)
+    )
+    .addUserOption((option) =>
+      option
+        .setName('user')
+        .setDescription(translateLanguage('myPoints.userOption'))
+        .setRequired(false)
+    )
+    .addStringOption((option) =>
+      option
+        .setName('channels')
+        .setDescription(translateLanguage('myPoints.channelsOption'))
+        .setRequired(false)
+    ),
+
+  async execute(interaction) {
+    try {
+      await interaction.deferReply({ ephemeral: true });
+
+      const currentDate = new Date();
+      const year =
+        interaction.options.getInteger('year') || currentDate.getFullYear();
+      const month =
+        interaction.options.getInteger('month') || currentDate.getMonth() + 1;
+      const user = interaction.options.getUser('user') || interaction.user;
+
+      const channelsInput = interaction.options.getString('channels');
+      const channels = channelsInput
+        ? channelsInput.split(',').map((channel) => channel.trim())
+        : [];
+
+      const targetStartDate = new Date(year, month - 1, 1);
+      const targetEndDate = new Date(year, month, 0);
+
+      const fetchedPoints = {
+        taskCompleted: 0,
+        addPoint: 0,
+        boostedPoint: 0,
+      };
+
+      function calculatePoints(messages, user, tagIds, fetchedPoints) {
+        messages.forEach((message) => {
+          if (message.author.id === user.id) {
+            if (message.content.includes(`<@&${tagIds.taskCompletedTagId}>`)) {
+              fetchedPoints.taskCompleted += 1;
+            }
+            if (message.content.includes(`<@&${tagIds.addPointTagId}>`)) {
+              fetchedPoints.addPoint += 1;
+            }
+            if (message.content.includes(`<@&${tagIds.boostedPointTagId}>`)) {
+              fetchedPoints.boostedPoint += 1;
+            }
+          }
+        });
+      }
+
+      async function processChannel(channelName) {
+        const channel = interaction.guild.channels.cache.find(
+          (ch) => ch.name === channelName && ch.isTextBased()
+        );
+        if (!channel) {
+          return;
+        }
+
+        const threads = await channel.threads.fetchActive();
+        for (const thread of threads.threads.values()) {
```

Comment:
It looks weird `threads.threads`, maybe changing the name it is more readable
```suggestion
        const activeThreads = await channel.threads.fetchActive();
        for (const thread of activeThreads.threads.values()) {
```